zap/src/config.rs

```diff
@@ -230,29 +230,29 @@ impl<'src> Ty<'src> {
 				if let Some(exact) = len.exact() {
 					(exact as usize, Some(exact as usize))
 				} else {
-					let len_size = len.numty().unwrap_or(NumTy::U16).size();
+					let len_numty = len.numty().map(|(numty, ..)| numty).unwrap_or(NumTy::U16);
 
 					(
-						len.min().map(|min| min as usize).unwrap_or(0) + len_size,
-						len.max().map(|max| (max as usize) + len_size),
+						len.min().map(|min| min as usize).unwrap_or(0) + len_numty.size(),
+						len.max().map(|max| max as usize + len_numty.size()),
 					)
 				}
 			}
 
 			Self::Arr(ty, len) => {
 				let (ty_min, ty_max) = ty.size(tydecls, recursed);
 				let len_min = len.min().map(|min| min as usize).unwrap_or(0);
-				let len_size = len.numty().unwrap_or(NumTy::U16).size();
+				let len_numty = len.numty().map(|(numty, ..)| numty).unwrap_or(NumTy::U16);
 
 				if let Some(exact) = len.exact() {
 					(ty_min * (exact as usize), ty_max.map(|max| ty_max.unwrap() * max))
-				} else if let Some(len_max) = len.max() {
+				} else {
 					(
-						ty_min * len_min + len_size,
-						ty_max.map(|ty_max| ty_max * (len_max as usize) + len_size),
+						ty_min * len_min + len_numty.size(),
+						ty_max
+							.and_then(|ty_max| len.max().map(|max| (ty_max, max as usize)))
+							.map(|(ty_max, max)| ty_max * max + len_numty.size()),
```

Comment:
A little nit - we can simplify the logic a little here so we don't have to map twice:
```suggestion
						ty_max
							.zip(len.max())
							.map(|(ty_max, max)| ty_max * max as usize + len_numty.size()),
```